spec/controllers/checkouts_controller_spec.rb

```diff
@@ -0,0 +1,106 @@
+require 'rails_helper'
+
+RSpec.describe CheckoutsController, type: :controller do
+  let(:user) { create(:user) }
+  let(:order_item) { create(:order_item) }
+  let(:order) { create(:order, order_items: [order_item]) }
+  let(:delivery) { create(:delivery) }
+
+  before do
+    allow(controller).to receive(:current_order).and_return(order)
+    allow(controller).to receive(:current_user).and_return(user)
+    sign_in(user)
+  end
+
+  describe '#show' do
+    context 'with :address step' do
+      it 'updates order status to in_progress' do
+        get :show, params: { id: :address }
+        expect(order.status).to eq('in_progress')
+      end
+    end
+
+    context 'with :delivery step' do
+      it 'renders next step' do
+        create(:billing, user: user)
```

Comment:
replace this to `let(:user) {crete(:user, :with_delivery)` and add `:with_delivery` trait to your UserFactory